src/btree/bt_delete.c

```diff
@@ -194,11 +194,19 @@ __wti_delete_page(WT_SESSION_IMPL *session, WT_REF *ref, bool *skipp)
      */
     WT_ERR(__wt_page_parent_modify_set(session, ref, false));
 
-    WT_ERR(__wt_calloc_one(session, &ref->page_del));
-
-    /* History store truncation is non-transactional. */
-    if (!WT_IS_HS(session->dhandle))
+    /*
+     * History store truncation is non-transactional, and fast truncate in history store applies
+     * only to globally visible pages. Therefore, ensure `page_del` remains null. (`page_del` being
+     * null indicates that the fast truncated page is globally visible.)
+     *
+     * An exception is selective backup, which can truncate non-globally visible history store
+     * pages. However, since this data is intended for permanent discard, it can also be treated as
+     * globally visible without causing issues.
+     */
+    if (!WT_IS_HS(session->dhandle)) {
+        WT_ERR(__wt_calloc_one(session, &ref->page_del));
```

Comment:
Why do we need to allocate a page del structure here? What does it do

Reply:
**Explanation of `page_del` allocation and its relevance to non-HS pages**

- `page_del` struct is allocated to hold details about deleted pages, including transaction ID, timestamp, and commit status. This information is essential for visibility checks, an also allowing us to determine if the deletion is committed during reconciliation.
- The `WT_REF_DELETED` flag alone marks a page as deleted, but `page_del` provides additional visibility details needed for reconciling/reading the deleted page accurately.
- For HS, `page_del` is unnecessary since HS is non-transaction the truncated data does not require transactional state tracking.
- A `null` `page_del` implies that the page is globally visible, allowing a faster decision for page removal during reconciliation. This assumption always applies to HS as part of this fix.

Reply:
I see thank you for the explanation